src/core/toga/widgets/button.py

```diff
@@ -11,21 +10,21 @@ class Button(Widget):
         id (str): An identifier for this widget.
         style (:obj:`Style`): An optional style object. If no style is provided then
             a new one will be created for the widget.
-        on_press (:obj:`callable`): Function to execute when pressed.
         enabled (bool): Whether or not interaction with the button is possible, defaults to `True`.
         factory (:obj:`module`): A python module that is capable to return a
             implementation of this class with the same name. (optional & normally not needed)
     """
 
-    def __init__(self, label, id=None, style=None, on_press=None, enabled=True, factory=None):
-        super().__init__(id=id, enabled=enabled, style=style, factory=factory)
+    def __init__(self, label, id=None, style=None, enabled=True, factory=None, **kwargs):
+        super().__init__(id=id, enabled=enabled, style=style, factory=factory, **kwargs)
 
         # Create a platform specific implementation of a Button
         self._impl = self.factory.Button(interface=self)
 
         # Set all the properties
         self.label = label
-        self.on_press = on_press
+
+    on_press = Event('Called when the button is pressed')
```

Comment:
It won't affect the execution of the code, but as a coding convention, I think it would make sense to have event declarations be the first thing in the class (same as you would attribute declarations in a Django class).

Reply:
Yeah, but does it make sense to place it between the docstring and `__init__`? Can/should I place it before the docstring? 

I simply went with what seemed to me to be the safest option, but LMK what you'd prefer and I'll change this.

Reply:
There's a difference between the *class* docstring and the `__init__` method's docstring. We've been a bit lax with this distinction historically; this seems like as good an opportunity as any to rectify the situation.

With that clarification, I'd say the event declarations go between the class docstring and the `__init__` docstring.

Reply:
I separated the docstrings and mover the event definition, please have another look before I apply this across the board to all classes.